Feature/auth frontend

LarsSonke opened the pull request.

LarsSonke committed:
feat: Add authentication and authorization frontend

- Add login and registration components with Material Design
- Implement JWT authentication service with token management
- Add auth guard for route protection
- Add HTTP interceptor for automatic token injection
- Create user models matching backend API (email, name, access_token)
- Add user profile menu in sidebar with logout functionality
- Update environment config to point to auth-service (port 8080)
- Add comprehensive authentication documentation

LarsSonke committed:
docs: Update README with authentication frontend info

LarsSonke committed:
refactor: Update auth components to modern Angular style and add forgot password

- Use inject() for dependency injection instead of constructor
- Use signal() for reactive state management
- Rename component files (remove .component suffix)
- Update templates to use signal() syntax
- Add forgot password feature with email input
- Add password reset functionality to auth service
- Add 'Forgot password?' link to login page
- Add /forgot-password route

LarsSonke committed:
feat: add password reset functionality

- Added reset-password component with token validation
- Implemented password confirmation validation
- Added resetPassword method to AuthService
- Configured route for /reset-password with token query param
- Auto-redirect to login after successful password reset

JulienNout commented:
De login zelf ziet er goed uit, maar zou deze niet in de content van de applicatie zetten.

Simon-Manasse commented on src/app/features/auth/register/register.html:
```diff
@@ -0,0 +1,102 @@
+          <mat-label>Full Name</mat-label>
+          <input matInput formControlName="name" autocomplete="name">
+          <mat-icon matPrefix>person</mat-icon>
+          @if (registerForm.get('name')?.hasError('required') && registerForm.get('name')?.touched) {
```
Hey, ik zie dat er in de template @if wordt gebruikt. Dat werkt helaas niet in Angular. In Angular gebruik je *ngIf voor conditionele weergave.

```
<!-- Onjuist -->
@if (registerForm.get('name')?.hasError('required') && registerForm.get('name')?.touched)
  <mat-error>Naam is verplicht</mat-error>
@endif

<!-- Correct -->
<mat-error *ngIf="registerForm.get('name')?.hasError('required') && registerForm.get('name')?.touched">
  Naam is verplicht
</mat-error>
```

JulienNout replied:
JulienNout reacted with hooray emoji

Simon-Manasse replied:
Ik denk niet dat het zo werkt man

LarsSonke committed:
feat: update login and register UI with modern styling and animations

- Enhanced login form with professional styling and animations
- Updated register form with improved layout and user experience
- Added smooth transitions and hover effects
- Improved form validation styling
- Consistent design language across auth components

LarsSonke committed:
fix: make auth pages standalone without sidebar and fix Angular template syntax
